src/test/java/com/agnieszkapawska/flashcards/domain/facades/FlashcardFacadeTest.java

```diff
@@ -0,0 +1,111 @@
+package com.agnieszkapawska.flashcards.domain.facades;
+
+import com.agnieszkapawska.flashcards.FlashcardAndQuestionTagAbstractTests;
+import com.agnieszkapawska.flashcards.domain.dtos.FlashcardSaveDto;
+import com.agnieszkapawska.flashcards.domain.dtos.FlashcardSaveResponseDto;
+import com.agnieszkapawska.flashcards.domain.models.Flashcard;
+import com.agnieszkapawska.flashcards.domain.models.QuestionTag;
+import com.agnieszkapawska.flashcards.domain.services.FlashcardService;
+import com.agnieszkapawska.flashcards.domain.services.QuestionTagService;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import java.util.*;
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.*;
+
+public class FlashcardFacadeTest extends FlashcardAndQuestionTagAbstractTests {
+    @Autowired
+    private FlashcardFacade flashcardFacade;
+    @MockBean
+    private FlashcardService flashcardService;
+    @MockBean
+    private QuestionTagService questionTagService;
+    private Flashcard flashcard;
+
+    @Before
+    public void createFlashcard() {
+        flashcard = new Flashcard();
+        flashcard.setId(1L);
+        flashcard.setQuestion("question");
+        flashcard.setAnswer("answer");
+        flashcard.setExplanation("explanation");
+        //questionTag with flashcards
+        QuestionTag questionTag = new QuestionTag();
+        questionTag.setName("home");
+        questionTag.setFlashcards(new HashSet<>(Collections.singletonList(new Flashcard("kubek", "mug"))));
+        flashcard.getQuestionTagsSet().add(questionTag);
+    }
+
+    @Test
+    public void saveFlashcard_ShouldReturnFlashcardSaveResponseDto() {
+        //given
+        FlashcardSaveDto flashcardSaveDto = new FlashcardSaveDto();
+
+        Set<QuestionTag> questionTagSet = new HashSet<>(Arrays.asList(createQuestionTag(1L, "home"),
+                createQuestionTag(2L, "holiday")));
+        flashcardSaveDto.setTagsSet(new HashSet<>(Arrays.asList("home", "holiday")));
+
+        when(flashcardService.saveFlashcard(any(Flashcard.class)))
+                .thenReturn(flashcard);
+        when(questionTagService.getQuestionTagsSet(anySet()))
+                .thenReturn(questionTagSet);
+
+        Long expectedId = 1L;
+        //when
+        FlashcardSaveResponseDto flashcardSaveResponseDto = flashcardFacade.saveFlashcard(flashcardSaveDto);
+        List<QuestionTag> questionTagList = new ArrayList<>(questionTagSet);
+        //then
+        Assert.assertNotNull(flashcardSaveResponseDto);
+        Assert.assertEquals(expectedId, flashcardSaveResponseDto.getId());
```

Comment:
You could also assert that `flashcardSaveResponseDto.getQuestionTagsSet` is equal to `questionTagList`

Reply:
not really because flashcardSaveResponseDto doesn't have questionTagList field